mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/property/OptionsModelProperty.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.property;
+
+import static java.util.stream.Collectors.toList;
+import org.mule.runtime.api.meta.model.ModelProperty;
+import org.mule.runtime.api.meta.model.declaration.fluent.ParameterDeclaration;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class OptionsModelProperty implements ModelProperty {
```

Comment:
other than consistency with the Metadata approach, is there a reason why this should be a model property instead of being explicit in the Model?